### PR TITLE
Added optional schema

### LINKED ISSUE #7: Default schema for the migrations table
Currently the public schema is the default for creating the migrations table, there doesnt seem to be a way to create it elsewhere.

I'd be tempted to say it is a bad idea because:
1. you may have another migrations table inside that public schema and it leaves you no options.
2. running execute on a schema where there is no migrations table will fail

What do you think of this: asyncpg-trek creates its own schema to handle migrations, and uses a migrations table inside it, low chances of conflicts, and you are able to run migrations on others schema, asyncpg-trek always knows where to find its migrations table

### PR DESCRIPTION
**Pull request by euri10**
Fixes #7 
Not sure I add a note in the readme about this

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ repos:
       args: ["--autofix"]
     - id: check-merge-conflict
   - repo: https://github.com/pycqa/isort
-    rev: "5.10.1"
+    rev: "5.11.5"
```
Comment (euri10) [inline .pre-commit-config.yaml:12]: see https://stackoverflow.com/a/75295105/3581357

Comment (adriangb) [reply .pre-commit-config.yaml:12]: Probably should just switch to ruff at some point…

File: asyncpg_trek/asyncpg.py [hunk 2]
```diff
@@ -6,58 +6,59 @@
...
-
 
 class AsyncpgExecutor:
-    def __init__(self, connection: asyncpg.Connection) -> None:
+    def __init__(self, connection: asyncpg.Connection, schema: str) -> None:
         self.connection = connection
+        self.schema = schema
 
     async def create_table_idempotent(self) -> None:
+        CREATE_TABLE = f"""\
```
Comment (adriangb) [inline asyncpg_trek/asyncpg.py]: Can we keep this at the module level and just call .format()?

File: asyncpg_trek/asyncpg.py [hunk 3]
```diff
@@ -6,58 +6,59 @@
...
         await self.connection.execute(RECORD_REVISION, from_revision, to_revision)  # type: ignore
 
     async def execute_operation(self, operation: Operation[asyncpg.Connection]) -> None:
         await operation(self.connection)
 
 
 class AsyncpgBackend:
-    def __init__(self, connection: asyncpg.Connection) -> None:
+    def __init__(
+        self, connection: asyncpg.Connection, schema: Optional[str] = "public"
```
Comment (adriangb) [inline asyncpg_trek/asyncpg.py]: Is it really optional? I’d think we want `schema: str = “public”`

**Review by adriangb — CHANGES_REQUESTED** [review]

File: asyncpg_trek/asyncpg.py [hunk 4]
```diff
@@ -7,57 +7,59 @@
 from asyncpg_trek._types import Operation
 
 CREATE_TABLE = """\
-CREATE TABLE IF NOT EXISTS migrations (
+CREATE SCHEMA IF NOT EXISTS {schema};
+CREATE TABLE IF NOT EXISTS {schema}.migrations (
```
Comment (adriangb) [inline asyncpg_trek/asyncpg.py]:
```suggestion
CREATE TABLE IF NOT EXISTS "{schema}.migrations" (
```
Mind adding quotes to these if you have a minute? It avoids issues with weird table / schema names.

**Review by adriangb — APPROVED** [review]

File: asyncpg_trek/asyncpg.py (hunk 3 above)
Comment (euri10) [reply asyncpg_trek/asyncpg.py]: nice one, reminds me of a gig where they had all their tables uppercase in postgres, that drove me nuts...

**Comment by euri10** [comment]
interesting CI fails, I commited to fast, quotes must be around table name and schema I think, will look at it later

**Pull request merged**